### PR TITLE
First step towards allowing polymorphic commands.

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
This change replaces `BedrockCommand` with `unique_ptr<BedrockCommand>` throughout Bedrock.

This probably helps to some minor degree with performance (Bedrock commands are never moved or copied), but that's not the main point of it.

#### It will have several benefits that will be realized in follow-up PRs

Plugins create their own commands. This means that instead of this sort of thing:
https://github.com/Expensify/Bedrock/blob/45079a66cc3d52623594af04f1971081e43f3387/BedrockCore.cpp#L99-L108

A plugin will either return a new command, which will implicitly mean that `peekedBy` and `processedBy` will be the given plugin, etc. Or it will return a null pointer, indicating that this plugin can't handle this command.

#### Because a plugin either returns a command object, or it doesn't, it means that there's no potential weirdness where one plugin could feasibly peek a command and another processes it. The plugin that creates the command will peek and process it.

Also, for places in Bedrock that need to look up data about plugins for commands:
https://github.com/Expensify/Bedrock/blob/45079a66cc3d52623594af04f1971081e43f3387/BedrockServer.cpp#L1639-L1657

The command knows all the info about its plugin and bedrock doesn't need to look this up again.

But the *main* reason to do this is:

### When `peek` and `process` are instance methods on specific command types, state can persist from `peek` to `process`inside the command

We've sort of done this in a hacky, weird way now with this code:
https://github.com/Expensify/Bedrock/blob/45079a66cc3d52623594af04f1971081e43f3387/BedrockCommand.h#L111-L119

Which ends up being used like this:
https://github.com/Expensify/Server-Expensify/blob/master/auth/AuthCommand.cpp#L12-L20

And:
https://github.com/Expensify/Server-Expensify/blob/master/auth/AuthCommand.cpp#L41-L45

Which is confusing and not very maintainable.

Instead, we'll be able to do something like:

```
class AuthCommand : public BedrockCommand {
};

class BillFund : public AuthCommand {
    int accountID = 0;

    peek() {
        accountID = request.calc64("accountID");
    }

    process() {
        // Account ID persists from peek to process.
        NVP::set(accountID, "name", "value");
    }
};
```

This change doesn't do any of the above, it's a preparatory refactor for those changes. But even then we have *less* after this change than before, which is usually a move in the right direction.

This is intended to pass existing tests and have no functional changes yet.

## Tests

Existing automated tests pass.

Existing auth tests pass.

### CODE REVIEW

**Comment by tylerkaraszewski** [comment]
Anyone know how to restart a build? @coleaeason @righdforsa? The restart button is missing in Travis for me.

**Comment by rafecolton** [comment]
@tylerkaraszewski restarted it

File: BedrockServer.cpp [hunk 1]
```diff
@@ -2142,16 +2148,16 @@ void BedrockServer::_acceptSockets() {
     }
 }
 
-void BedrockServer::waitForHTTPS(BedrockCommand&& command) {
+void BedrockServer::waitForHTTPS(unique_ptr<BedrockCommand>&& command) {
     lock_guard<mutex> lock(_httpsCommandMutex);
 
-    // Create a new BedrockCommand on the head via moving from our existing command. This is the one we'll store.
-    BedrockCommand* commandPtr = new BedrockCommand(move(command));
+    // Un-uniquify the unique_ptr. I don't love this, but it works better with the code we've already got.
```
Comment (coleaeason) [inline BedrockServer.cpp]: what's the alternative here?

Comment (tylerkaraszewski) [reply BedrockServer.cpp]: The alternative is to combine `_outstandingHTTPSCommands` and `_outstandingHTTPSRequests` into one data structure that stores only a single copy of the command pointer, but is searchable by both command pointer and transaction. Which is doable but obviously more work.

Comment (coleaeason) [reply BedrockServer.cpp]: can we do that as a follow up? :D

**Pull request merged**